### ISSUE TITLE
redirect_stdout() hangs on Windows

### ISSUE THREAD

**Issue by mpastell** [JuliaLang/julia#19194]
The following code hangs permanently on Windows 10 x64 using Julia 0.5 prebuilt binary

```julia
oldSTDOUT = STDOUT
rd, rw = redirect_stdout()

print(Array(1:10))

redirect_stdout(oldSTDOUT)
close(rw)
res = readstring(rd)
close(rd)

println(res)
```

**Comment by yuyichao** [comment]
I think this is expected. You need to have a reader before closing the stream.

**Comment by mpastell** [comment]
Do you have an example? This used to work on 0.4 and runs fine on Ubuntu.

**Comment by vtjnash** [comment]
The Linux kernel is better at buffering than the windows NT kernel. But that's just an implementation detail. This used to work on v0.4, but didn't actually guarantee your data would get written. That's been fixed now in v0.5.

**Comment by mpastell** [comment]
Thanks, I think I figured it out:

```julia
oldSTDOUT = STDOUT
rd, rw = redirect_stdout()
reader = @async readstring(rd)
print(Array(1:10))

redirect_stdout(oldSTDOUT)

close(rw)
res = wait(reader)
close(rd)

println(res)
```
